ossfuzz/fuzz_data_producer.h

```diff
@@ -0,0 +1,27 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+FUZZ_STATIC uint32_t FUZZ_produceUint32Range(uint8_t *data, size_t size,
```

Comment:
I think we need to make this a stateful struct and take ownership of the buffer. Otherwise two calls to this function will produce the same data. We want to use this API for the rest of the fuzzers too. For example, we'll need to implement [FUZZ_randomPreferences()](https://github.com/lz4/lz4/blob/dev/ossfuzz/lz4_helpers.h#L8) using the `FUZZ_dataProducer_t` instead of the seed.

```c
// Header
typedef struct {
  uint8_t* data;
  size_t size;
} FUZZ_buffer_t;

typedef struct FUZZ_dataProducer_s FUZZ_dataProducer_t;
FUZZ_dataProducer_t* FUZZ_dataProduer_create(uint8_t* data, size_t size);
void FUZZ_dataProducer_free(FUZZ_dataProducer_t* producer);
uint32_t FUZZ_dataProducer_uint32(FUZZ_dataProducer_t* producer, uint32_t min, uint32_t max);
FUZZ_buffer_t FUZZ_dataProducer_remaining(FUZZ_dataProducer_t* producer);
// Whatever other functions you need...

// C file
struct FUZZ_producer_s {
  uint8_t* data;
  size_t size;
};
```